oauthproxy.go

```diff
@@ -531,39 +563,7 @@ func (p *OAuthProxy) IsValidRedirect(redirect string) bool {
 }
 
 func (p *OAuthProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
-	p.preAuthChain.Then(http.HandlerFunc(p.serveHTTP)).ServeHTTP(rw, req)
-}
-
-func (p *OAuthProxy) serveHTTP(rw http.ResponseWriter, req *http.Request) {
-	if req.URL.Path != p.AuthOnlyPath && strings.HasPrefix(req.URL.Path, p.ProxyPrefix) {
-		prepareNoCache(rw)
-	}
-
-	switch path := req.URL.Path; {
-	case path == p.RobotsPath:
-		p.RobotsTxt(rw, req)
-	case p.IsAllowedRequest(req):
-		p.SkipAuthProxy(rw, req)
```

Comment:
Where did this case go? I'm not seeing it readily above in the new router setup

Reply:
This is the reason why this PR fixes #992 - rather than making a special case for skipped routes/pre-approved source IPs and passing straight through to the upstream without attempting to load the session, this PR rearranges things so that _all_ requests go through the `sessionChain` first and then the `IsAllowedRequest` check happens in `getAuthenticatedSession`.

The logic has changed from

  - if IsAllowedRequest
    - then proxy
    - else
      - try loading session
      - if session was loaded successfully
        - then proxy
        - else redirect to signin

to

  - try loading session
  - if IsAllowedRequest _or_ session was loaded successfully
    - then proxy
    - else redirect to signin

which has the side effect of adding headers whenever there's a valid session available, even if the route is skipped or the source IP is allowlisted.

Reply:
Ah thanks! I see the flow now